python/torch_mlir/dialects/torch/importer/jit_ir/csrc/torch_to_mlir_utils.cpp

```diff
@@ -407,15 +409,52 @@ MlirAttribute torch_mlir::importAttribute(MlirLocation loc,
 
 MlirLocation torch_mlir::getMlirLocationFromNode(MlirContext context,
                                                  torch::jit::Node *node) {
-  auto flc = node->sourceRange().file_line_col();
-  if (flc) {
+  MlirLocation loc = mlirLocationUnknownGet(context);
+
+  auto getLocOpName = [&](const c10::FunctionSchema * schema)->MlirLocation {
+    return mlirLocationNameGet(context,
+                               toMlirStringRef(schema->operator_name().name),
+                               /*childLoc=*/{nullptr});
+  };
+
+  if (node->hasAttribute(c10::Symbol::attr("source_files"))) {
```

Comment:
So, the current logic is:
 1. Try to get the full stack trace. If found, the CallSite is the only loc and the op name is skipped.
 2. Else, try to get the single flie/line/col, and add a `FusedLoc[operator_name, FileLineCol]`
 3. Otherwise, just use the op name.
Finally, always name the location with the scope if it exists.

That's slightly inconsistent and a creative interpretation of `FusedLoc`. Also, it might involve a `NameLoc(NameLoc)` if no source code location is present which is technically legal in newer MLIR versions, but a bit strange and also won't be parsable by older MLIR versions I happen to know you're interested in.

Instead, I'd suggest a 2 stage approach, with the idea that if we have name scopes, the simpler op name location may not provide any additional info:

First:
1. Try to get `loc = CallsiteLoc(...)`
2. Else try get just `loc = FileLineColLoc(...)`
3. Else, `loc = UnknownLoc`

Next, add the relevant "name", with the source code loc as the child:
1. Try to get the scope, and add `loc = Name(scope, loc)`
2. Else, fall back on the op name and add `loc = Name(operator_name, loc)`

Alternatively, we can do what pytorch_xla does and combine them into one "name", something like:
```
name = operator_name
if scope:
   name = scope + "/" + name
```

Reply:
> Try to get the scope, and add loc = Name(scope, loc)
Else, fall back on the op name and add loc = Name(operator_name, loc)

That looks reasonable! Just pushed a commit to match new logic. Please take a look.

Reply:
CC: @ramiro050

Reply:
So we actually just discovered a use case where it would be helpful to have the oeprator name _and_ the other location info (scope + stack trace), to handle performance profiilng per-op-basis. 

If the user has pytorch code like
```python
a = torch.add(b,c).mul(d,e)
```

Then both ops will have the same location.

If we go with the `{scope}/{op_name}` route, they'll be distinct.

Reply:
Thanks @mabrowning for the help on this!